### PR TITLE
PICO checker

### PR DESCRIPTION
**Pull request by Ao-senXiong**
I am not sure this will be easier or harder to review for PICO. I can getting PICO to eisop since ultimately we will do.

### CODE REVIEW

**Comment by wmdietl** [comment]
Let's finish reviewing and improving this in the separate repo before moving it here.

File: build.gradle [hunk 1]
```diff
@@ -256,6 +256,8 @@ allprojects {
                 '**/returnsreceiverdelomboked/*',
                 '**/build/**',
                 '*/dist/**',
+                // Don't format util the error report location is fixed
```
Comment (wmdietl) [inline build.gradle:259]: All tests should be formatted properly. There is nothing special about PICO tests.

File: checker-qual/src/main/java/org/checkerframework/checker/immutability/qual/Assignable.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD})
+@HoldsForDefaultValue
+public @interface Assignable {}
```
Comment (wmdietl) [inline checker-qual/src/main/java/org/checkerframework/checker/immutability/qual/Assignable.java:15]: Please add javadoc for everything (again, in the PICO repo).

File: checker/bin-devel/test-misc.sh [hunk 3]
```diff
@@ -1,67 +1,67 @@
...
-     echo "docs/manual/contributors.tex is not up to date." &&
-     echo "If the above suggestion is appropriate, run: make -C docs/manual contributors.tex" &&
-     echo "If the suggestion contains a username rather than a human name, then do all the following:" &&
-     echo "  * Update your git configuration by running:  git config --global user.name \"YOURFULLNAME\"" &&
-     echo "  * Add your name to your GitHub account profile at https://github.com/settings/profile" &&
-     echo "  * Make a pull request to add your GitHub ID to" &&
-     echo "    https://github.com/eisop-plume-lib/plume-scripts/blob/master/git-authors.sed" &&
-     echo "    and remake contributors.tex after that pull request is merged." &&
-     false)
+##!/bin/bash
```
Comment (wmdietl) [inline checker/bin-devel/test-misc.sh:1]: Why are all the misc tests disabled?

File: checker/src/main/java/org/checkerframework/checker/immutability/ExtendedViewpointAdapter.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.checkerframework.checker.immutability;
+
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.ViewpointAdapter;
+
+import javax.lang.model.element.AnnotationMirror;
+
+public interface ExtendedViewpointAdapter extends ViewpointAdapter {
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/immutability/ExtendedViewpointAdapter.java:8]: We really need some documentation for what this interface is for. Is it PICO specific? Should it just be in the main interface?

**Review by wmdietl — COMMENTED** [review]
Some small comments, to address in the PICO repo.

File: checker/src/main/java/org/checkerframework/checker/immutability/ExtendedViewpointAdapter.java (hunk 4 above)
Comment (Ao-senXiong) [reply checker/src/main/java/org/checkerframework/checker/immutability/ExtendedViewpointAdapter.java:8]: @wmdietl I removed those dummy interfaces in [2789ede](https://github.com/opprop/immutability/pull/43/commits/2789edea849c3897ff0ec2c0524b8127fa3da20a)

File: build.gradle (hunk 1 above)
Comment (Ao-senXiong) [reply build.gradle:259]: Yes, that's the next thing we should look at in the future. I opened an issue in opprop https://github.com/opprop/immutability/issues/53.

File: checker-qual/src/main/java/org/checkerframework/checker/immutability/qual/Assignable.java (hunk 2 above)
Comment (Ao-senXiong) [reply checker-qual/src/main/java/org/checkerframework/checker/immutability/qual/Assignable.java:15]: Sure, I will create a new PR to add the Javadoc.

File: checker/bin-devel/test-misc.sh (hunk 3 above)
Comment (Ao-senXiong) [reply checker/bin-devel/test-misc.sh:1]: A lot of Javadoc issues and I just disabled for testing other tasks.

Comment (wmdietl) [reply checker/bin-devel/test-misc.sh:1]: @Ao-senXiong For the future: to disable the Javadoc checks, do `touch SKIP-REQUIRE-JAVADOC` in the top-level directory and add that file.

Comment (Ao-senXiong) [reply checker/bin-devel/test-misc.sh:1]: Sure, thanks.

**Pull request closed without merging**